RecoLocalCalo/EcalDeadChannelRecoveryAlgos/src/EcalDeadChannelRecoveryAlgos.cc

```diff
@@ -8,30 +8,41 @@
 //      Feb 14 2013:   Implementation of the criterion to select the "correct"
 // max. cont. crystal.
 //
+//modified by S.Taroni, N. Marinelli 11 June 2019
 
 #include "RecoLocalCalo/EcalDeadChannelRecoveryAlgos/interface/EcalDeadChannelRecoveryAlgos.h"
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 
 template <typename T>
-void EcalDeadChannelRecoveryAlgos<T>::setCaloTopology(const CaloTopology *topo) {
-  nn.setCaloTopology(topo);
+void EcalDeadChannelRecoveryAlgos<T>::setParameters(const edm::ParameterSet &ps) {
+  bdtg_.setParameters(ps);
 }
 
 template <typename T>
-EcalRecHit EcalDeadChannelRecoveryAlgos<T>::correct(
-    const T id, const EcalRecHitCollection &hit_collection, std::string algo, double Sum8Cut, bool *AcceptFlag) {
-  // recover as single dead channel
-  double NewEnergy = 0.0;
+void EcalDeadChannelRecoveryAlgos<T>::setCaloTopology(std::string algo, const CaloTopology *topo) {
+  bdtg_.setCaloTopology(topo);
```

Comment:
In a way it's vestigial because it was used when the two recovery algorythms were coexisting. Although we have decided to throw away the NN, we do not want to remove the possibility for a future to include an other algo

Reply:
but what would be the reason to keep it here, after the choice of the algo is made?
Even in the older variant of the code where both NN and BDTG were (unnecessarily) instantiated together there was still an assumption that only one of them can run.
If you really need it, the choice of the algo should be done when the EcalDeadChannelRecoveryAlgos is constructed, not in this method.
So, I do not see a reason for the algo parameter here.

Reply:
The choice is made in https://github.com/nancymarinelli/cmssw/blob/deadChannelRecoveryBDT_V1/RecoLocalCalo/EcalRecProducers/plugins/EcalRecHitWorkerRecover.cc#L127 and the setCaloToplogy is necessary to pass the topology to the chosen algo. At this point in time, and with all the ifs related to the main validation, we do not certainly feel to make any change to  how the framework was setup

Reply:
right, I know this point of entry for the topology.
My comment still stands.
 the choice of the algorithm is already made earlier (in ```ebDeadChannelCorrector.setParameters(ps)```) there is no reason or a reasonable way to modify it here event by event.